Libraries/oneTBB/tbb-resumable-tasks-sycl/src/tbb-resumable-tasks-sycl.cpp

```diff
@@ -0,0 +1,137 @@
+//==============================================================
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <cmath>
+#include <array>
+#include <thread>
+#include <iostream>
+#include <atomic>
+#include <sstream>
+
+#include <CL/sycl.hpp>
+
+#include <tbb/blocked_range.h>
+#include <tbb/global_control.h>
+#include <tbb/task.h>
+#include <tbb/task_group.h>
+#include <tbb/parallel_for.h>
+
+constexpr cl::sycl::access::mode sycl_read = cl::sycl::access::mode::read;
+constexpr cl::sycl::access::mode sycl_write = cl::sycl::access::mode::write;
+
+const float ratio = 0.5; // CPU or GPU offload ratio
+const float alpha = 0.5; // coeff for triad calculation
+
+constexpr std::size_t array_size = 16;
+std::array<float, array_size> a_array; // input array
+std::array<float, array_size> b_array; // input array
+std::array<float, array_size> c_array; // output array
+
+void print_array( const char* text, const std::array<float, array_size>& array ) {
+    std::cout << text;
+    for (const auto& s: array) std::cout << s << ' ';
+    std::cout << std::endl;
+}
+
+class AsyncActivity {
+    float offload_ratio;
+    std::atomic<bool> submit_flag;
+    tbb::task::suspend_point suspend_point;
+    std::thread service_thread;
+
+public:
+    AsyncActivity() : offload_ratio(0), submit_flag(false),
+        service_thread([this] {
+            while(!submit_flag) {
+                std::this_thread::yield();
+            }
+            std::size_t array_size_sycl = std::ceil(array_size * offload_ratio);
+            std::stringstream sstream;
+            sstream << "start index for GPU = 0; end index for GPU = "
+                    << array_size_sycl << std::endl;
+            std::cout << sstream.str();
+            const float coeff = alpha; // coeff is a local variable
+
+            { // starting SYCL code
+                cl::sycl::range<1> n_items{array_size_sycl};
+                cl::sycl::buffer<cl_float, 1> a_buffer(a_array.data(), n_items);

```

Comment:
You should be using the simplified buffer allocation form.   Also the DPC++/SYCL 2020 prefix is sycl:: not cl::sycl::

Reply:
Could you please clarify, what is the simplified buffer allocation form? Other samples omits template parameters, but it will not work in C++11.

Reply:
Great news: SYCL 2020 requires at least C++17... :-)
But yes, I agree SYCL 1.2.1 would allow you to use a 10 year-old version of C++.
But who would like such an old environment in 2020? :-(